lib/bugzilla_service.rb

```diff
@@ -13,26 +13,126 @@ class BugzillaService
     resolved
   )
 
+  # Helper class for specific bugs
+  class Bug
+    attr_reader   :service
+    attr_accessor :id, :status
+
+    def initialize(service, id, status)
+      @service, @id, @status = service, id, status
+    end
+
+    def comments
+      response = service.get("/rest/bug/#{id}/comment") do |req|
+        req.params["include_fields"] = "text"
+      end
+
+      if response.status == 200
+        JSON.parse(response.body)["bugs"][id.to_s]["comments"]
+            .map { |comment| comment["text"] }
+      else
+        []
+      end
+    end
+
+    # Adds a comment.  Returns true if status == 200
+    def add_comment(text)
+      payload  = {"comment" => text}.to_json
+      response = service.post("/rest/bug/#{id}/comment", payload)
+
+      response.status == 200
+    end
+
+    # Basicially API compatible with `active_bugzilla` way of updating the bug
+    # status.
+    def save
+      payload  = { "ids" => [id], "status" => status }.to_json
+      response = service.put("/rest/bug/#{id}", payload)
+
+      response.status == 200
+    end
+  end
+
   def initialize
-    service # initialize the service
+    connection # initialize the connection
   end
 
-  def service
-    @service ||= begin
-      require 'active_bugzilla'
-      bz = ActiveBugzilla::Service.new(*credentials.to_h.values_at(:url, :username, :password))
-      ActiveBugzilla::Base.service = bz
+  def connection
+    @connection ||= begin
+      require 'faraday'
+
+      Faraday.new(:url => credentials.to_h[:url]) do |faraday|
+        faraday.request(:url_encoded)
+        faraday.adapter(Faraday.default_adapter)
+      end
     end
   end
 
   def find_bug(id)
-    ActiveBugzilla::Bug.find(:product => Settings.bugzilla.product, :id => id).first
+    response = get("/rest/bug") do |req|
+      product = Settings.bugzilla.product
+      req.params["id"]             = id
+      req.params["product"]        = product if product
```

Comment:
This is required, so I think we need to ensure product exists otherwise raise an exception.  We can't accidentally mess with bugs that aren't "ours"

Reply:
Roger, I will make it so.

![Make it so](https://media.giphy.com/media/bKnEnd65zqxfq/giphy.gif)